### PR TITLE
Rewrite CLI

### PR DESCRIPTION
**Pull request by moonso**
# Summary

This pull request reworks the CLI completely. The idea is to make it more clear how bundles, versions and files are linked together. Each subcommand will follow the same structure so there will for example housekeeper get

```bash
housekeeper get --help
Usage: housekeeper get [OPTIONS] COMMAND [ARGS]...

  Get info from database

Options:
  --help  Show this message and exit.

Commands:
  bundles  Get bundle from database
  files    Get files from database
  tags     Get the tags from database
  version  Get versions from database
```

and add:

```bash
housekeeper add --help
Usage: housekeeper add [OPTIONS] COMMAND [ARGS]...

  Add things to the store.

Options:
  --help  Show this message and exit.

Commands:
  bundle   Add a new bundle.
  file     Add a file to the latest version of a bundle.
  tag      Add tags to housekeeper.
  version  Add a new version to a bundle.
```

### How to prepare for test

- [x] ssh to hasta
- [x] install on stage:
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-housekeeper-stage.sh log-output`
- [x] Check that there is a mip cases to store (vitalmouse)

### How to test
- [x] Store a case in housekeeper (this is to check that the interaction between CG and hk works as expected)

### Expected test outcome
- [x] check that the test succeded
- [x] Take a screenshot and attach or copy/paste the output.

Skärmavbild 2020-09-14 kl  13 21 56

## Review
- [x] code approved by @Mropat 
- [x] tests executed by @moonso 
- [x] "Merge and deploy" approved by @moonso 
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [x] **MAJOR** - when you make incompatible API changes

### CODE REVIEW

**Comment by moonso** [comment]
Ok @Mropat this is ready for review!! 🌵 🛄 🌵

File: tests/store/test_store_api.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def test_fetch_no_files_before_oldest(
     """
     store = populated_store
     # GIVEN a store with two bundles and two files in each bundle
-    bundle_old_obj = store.add_bundle(data=bundle_data_old)[0]
+    bundle_old_obj, _ = store.add_bundle(data=bundle_data_old)
```
Comment (Mropat) [inline tests/store/test_store_api.py:51]: nice!

**Review by Mropat — APPROVED** [review]
Wonderful PR, this makes the CLI much easier to parse!
I had some very minor comments

File: housekeeper/cli/add.py [hunk 2]
```diff
@@ -1,111 +1,237 @@
...
 
 @click.group()
 def add():
     """Add things to the store."""
 
 
-@add.command()
-@click.argument("name")
+@add.command("bundle")
+@click.argument("bundle_data", required=False)
```
[the remark this answers is not in the document]
Comment (moonso) [reply housekeeper/cli/add.py:62]: Yes!

File: housekeeper/cli/add.py [hunk 3]
```diff
@@ -1,111 +1,237 @@
...
 import click
+from marshmallow.exceptions import ValidationError
+
+from housekeeper.date import get_date
+from housekeeper.store.api import schema as schemas
+
+LOG = logging.getLogger(__name__)
+
+
+def load_json(json_str: str) -> dict:
```
[the remark this answers is not in the document]
Comment (moonso) [reply housekeeper/cli/add.py:18]: Sure

[the remark this answers is not in the document]
Comment (moonso) [reply housekeeper/cli/add.py:18]: Fixed!

**Comment by patrikgrenfeldt** [comment]
Suggestion, please change "cannot delete X, X contains Y" to "Cannot delete X, please remove all Ys first" to let the user know what actions are needed

**Comment by moonso** [comment]
Skärmavbild 2020-09-15 kl  09 35 10

Merged and deployed

**Pull request merged**